w11h03/test/pgdp/networking/DataHandlerTest.java

```diff
@@ -0,0 +1,128 @@
+package pgdp.networking;
+
+import com.sun.net.httpserver.HttpServer;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DataHandlerTest {
+
+    DataHandler dataHandler;
+    HttpServer server;
+
+    private static final String USERNAME = "TestUser123";
+    private static final String ID = "geXYZtup";
+    private static final String PASSWORD = "verySecurePassw0rd";
+    private static final String DUMMY_TOKEN = "ijd3948rzjsdfgbß9834z";
+
+    @BeforeEach
+    public void setup() throws IOException {
+
+        server = HttpServer.create();
+        server.bind(new InetSocketAddress("127.0.0.1", 1337), 0);
+
+        HttpClient client = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .connectTimeout(Duration.ofSeconds(20))
+                .proxy(ProxySelector.of(new InetSocketAddress("127.0.0.1", 1337)))
+                .build();
+
+        dataHandler = new DataHandler();
+        dataHandler.setClient(client);
+
+        server.start();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        server.stop(0);
+    }
+
+    void registerRequestTest(Request actual, String username, String id) {
+        assertEquals("POST", actual.requestMethod());
+        assertEquals(List.of("application/json"), actual.requestHeaders().get("Content-Type"));
+        assertEquals(URI.create("http://carol.sse.cit.tum.de/api/users/register"), actual.requestURI());
```

Comment:
```suggestion
        assertEquals(URI.create("http://carol.sse.cit.tum.de/api/user/register"), actual.requestURI());
```
see http://131.159.47.114/docs